core/src/main/java/org/sonarsource/sonarlint/core/WsHelperImpl.java

```diff
@@ -80,61 +83,35 @@ static ValidationResult validateConnection(SonarLintWsClient client, @Nullable S
-    try {
-      // in 5.3 login is mandatory and user needs admin privileges
-      serverChecker.checkVersionAndStatus("5.4");
-
-      if (force) {
-        // revoke
-        client.post("api/user_tokens/revoke?name=" + name);
-      }
-
-      // create
-      try (WsResponse response = client.post("api/user_tokens/generate?name=" + name)) {
-        Map<?, ?> javaRootMapObject = new Gson().fromJson(response.content(), Map.class);
-        return (String) javaRootMapObject.get("token");
-      }
-    } catch (RuntimeException e) {
-      throw SonarLintWrappedException.wrap(e);
-    }
-  }
-
-  @Override
-  public String generateAuthenticationToken(ServerConfiguration serverConfig, String name, boolean force) {
-    SonarLintWsClient client = createClient(serverConfig);
-    return generateAuthenticationToken(new ServerVersionAndStatusChecker(client), client, name, force);
+  private SonarLintWsClient createClient(ConnectedModeEndpoint endpoint) {
+    requireNonNull(endpoint);
+    return new SonarLintWsClient(endpoint, client);
   }
 
   @Override
-  public List<RemoteOrganization> listUserOrganizations(ServerConfiguration serverConfig, @Nullable ProgressMonitor monitor) {
-    SonarLintWsClient client = createClient(serverConfig);
+  public List<RemoteOrganization> listUserOrganizations(ConnectedModeEndpoint endpoint, @Nullable ProgressMonitor monitor) {
+    SonarLintWsClient client = createClient(endpoint);
     ServerVersionAndStatusChecker serverChecker = new ServerVersionAndStatusChecker(client);
     return listUserOrganizations(client, serverChecker, new ProgressWrapper(monitor));
   }
 
   @Override
-  public Optional<RemoteOrganization> getOrganization(ServerConfiguration serverConfig, String organizationKey, @Nullable ProgressMonitor monitor) {
-    SonarLintWsClient client = createClient(serverConfig);
-    ServerVersionAndStatusChecker serverChecker = new ServerVersionAndStatusChecker(client);
-    return getOrganization(client, serverChecker, organizationKey, new ProgressWrapper(monitor));
+  public Optional<RemoteOrganization> getOrganization(ConnectedModeEndpoint endpoint, String organizationKey, @Nullable ProgressMonitor monitor) {
+    SonarLintWsClient wsClient = createClient(endpoint);
+    ServerVersionAndStatusChecker serverChecker = new ServerVersionAndStatusChecker(wsClient);
+    return getOrganization(wsClient, serverChecker, organizationKey, new ProgressWrapper(monitor));
   }
 
   @Override
-  public Optional<RemoteProject> getProject(ServerConfiguration serverConfig, String projectKey, ProgressMonitor monitor) {
-    SonarLintWsClient client = createClient(serverConfig);
-    return fetchComponent(client, projectKey)
+  public Optional<RemoteProject> getProject(ConnectedModeEndpoint endpoint, String projectKey, ProgressMonitor monitor) {
+    SonarLintWsClient wsClient = createClient(endpoint);
```

Comment:
Each time we have to create a WsClient. I wonder if it shouldn't be a constructor parameter.